### ISSUE TITLE
Hide adaptive modes (theme, brightness/redshift) if geoclue-2.0 DBus service is not available

### ISSUE THREAD

**Issue by sunweaver** [AyatanaIndicators/ayatana-indicator-display#95]
If there is not geoclue-2.0 provider on the DBus session bus, hide the adaptive modes in display indicator.

**Comment by tari01** [comment]
I don't think we need to do this: the indicator has a fallback longitude/latitude which can be tweaked in GSettings.

**Comment by sunweaver** [comment]
> I don't think we need to do this: the indicator has a fallback longitude/latitude which can be tweaked in GSettings.

Please hide those menu items if they cannot be used. Note that the customer's target system is an enterprise desktop. Showing features that don't exist is no option there.

**Comment by sunweaver** [comment]
So, for clarification:

For the enterprise Linux, we want:

  * theme switching
  * manual adjustment of brightness and redness

The enterprise desktop won't have geoclue running and in that case, the following menu items should be invisible/hidden:

  * color temperature -> adaptive (cooler|(normal)|warmer)
  * theme profile -> adaptive

The adaptive seems to be alluding auto-leveling those well you by some information (time of day, timezone, location, time of year).

Are you saying that the auto-leveling works if we simply roll-out geo coordinates via GSettings?

**Comment by sunweaver** [comment]
If so, we should be able to allow the user to adjust those geo-coordinates. (We must assume that they go travelling from time to time).

**Comment by sunweaver** [comment]
However, this might be beyond the scope of a display indicator. In fact, we have white papers for a location indicator. There this would be a nice feature to land.

Not sure how to handle this for now, though.

**Comment by tari01** [comment]
> Are you saying that the auto-leveling works if we simply roll-out geo coordinates via GSettings?

Yes. I have written it so that it never crashes regardless of the geolocation state. It starts up with the last coordinates stored in GSettings (Greenwich by default) and updates them if geoclue becomes available. If geoclue times out, it stays as-is.
